a2a/auth.go

```diff
@@ -14,8 +14,72 @@
 
 package a2a
 
-// Defines a security scheme that can be used to secure an agent's endpoints.
-// This is a discriminated union type based on the OpenAPI 3.0 Security Scheme.
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A declaration of the security schemes available to authorize requests. The key
+// is the scheme name. Follows the OpenAPI 3.0 Security Scheme Object.
+type NamedSecuritySchemes map[string]SecurityScheme
+
+func (s *NamedSecuritySchemes) UnmarshalJSON(b []byte) error {
+	var schemes map[string]json.RawMessage
+	if err := json.Unmarshal(b, &schemes); err != nil {
+		return err
+	}
+
+	result := make(map[string]SecurityScheme, len(schemes))
+	for k, v := range schemes {
+		type typedScheme struct {
+			Type string `json:"type"`
+		}
+		var ts typedScheme
+		if err := json.Unmarshal(v, &ts); err != nil {
+			return err
+		}
+
+		switch ts.Type {
+		case "apiKey":
+			var scheme APIKeySecurityScheme
+			if err := json.Unmarshal(v, &scheme); err != nil {
+				return err
```

Comment:
nit: does this err mention the &scheme's type? Otherwise, I think consider annotating the error with the ts.Type or schema type name.